### PR TITLE
Counter visualization: formatting not applied to target value

### PR DESCRIPTION
**Pull request by kravets-levko**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Bug Fix

## Description

Refactor:
- extracted formatting logic to a function (and revisited it) - so it will be easier later move it to db migration (added TODO to explain it);
- fixed some edge case handling (e.g. when target value column selected but it's not a number - previously counter highlighted "trend negative" value and showed an empty target value);
- replaced `underscore.string/numberFormat` with `numeral` and removed `underscore.string` dependency - it was the only file that used it;
- replaced AngularJS `number` filter with `numeral` (for consistency and to simplify future React migration); added `tooltipFormat` options that currently just matches `number` filter behavior; added TODO to implement this option in editor.

## Related Tickets & Documents

Fixes getredash/redash#4031

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

Before:
image

After:
image

### CODE REVIEW

**Comment by kravets-levko** [comment]
@susodapop You're welcome to test the fix as well :bowing_man:

**Comment by susodapop** [comment]
image
Yeah it works properly on my example!

File: client/app/visualizations/counter/index.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import numberFormat from 'underscore.string/numberFormat';
```
Comment (ranbena) [inline client/app/visualizations/counter/index.js]: Nice!

**Comment by ranbena** [comment]
> image
> Yeah it works properly on my example!

Damn @susodapop you don't got Roboto on your system. Until we fix it, can you install it on your OS (for the vids)?

**Comment by susodapop** [comment]
@ranbena You are way better at spotting fonts than I am! I downloaded Roboto to my system. Previous to that the counter was rendered in San Francisco.

image

**Pull request merged**